pymc3_hmm/step_methods.py

```diff
@@ -684,6 +703,8 @@ def __init__(self, vars, values=None, model=None):
     def step(self, point):
         X = self.X_fn(point)
         y, X = self.y_X_fn(point, X)
+        if isinstance(y, at.Variable):
+            y = y.eval()
```

Comment:
Shouldn't `self.y_X_fn` return non-symbolic values by definition?  The basic idea is that all these objects with the suffix `_fn` are supposed to be compiled Theano/Aesara functions that necessarily return numeric&mdash;and not symbolic&mdash;values.

Reply:
That is what I thought but if I put in y as a shared tensor, then it would not be complied.

Reply:
Shared variables can be included as outputs to `[aesara|theano].function`, but not inputs.

Reply:
Got it. Thanks,  I found the bug which is causing this.